Http api schema endpoint (#736)
arangodb/docs-hugo: site/content/3.12/develop/http-api/schemas.md

---

> title: HTTP interface for database structures (schema, graph, view)

Reviewer: This is ambiguous IMO. I generally wonder how to call it, because just Schemas can be confused with document schemas. Should we refer to it as "sampled schemas", "inferred schemas", or something similar?

menuTitle: Schemas
description: >-
   HTTP interface for database structures gives you information about 
   graphs, views, collections and schemas in the database.
# GET /_api/schema
# GET /_api/schema/graph/{graph-name}
# GET /_api/schema/view/{view-name}

> # GET /_api/schema/collection/{collection-name}

Author: I just thought I should write notes what endpoints are supported but I deleted them.

> 

Author: I set the `weight` to 33. `Documents` = 30, `Graphs` = 35.

---

The interface provides the means to collect information on database 
structures including graphs, views, collections and their schemas 
at one stop. This information is helpful if you want to understand 
the overall shape and structure of the database.

---

## What This Interface Returns
1. **graphs** – each graph shows its name and how it connects collections using edges (`_from` and `_to`).
2. **views** – each view shows its name and which collections and fields it links to.
3. **collections** – only connected or linked collections are shown. Each collection shows:
    - a list of attributes (fields)
    - the data types of each attribute (`string`, `number`, `bool`, or `object`)
    - whether the attribute is optional (meaning some documents/edges may not have it)

>     - and example documents or edges for reference

Reviewer: This should rather be described in a structured form in the per-endpoint OpenAPI description, and any explanation needed in addition can be put in the endpoint `description`.

Author: I moved these explanations to each endpoint, but they all say overlapping things. Is that okay? I assume users won’t read the entire document but will only look at specific endpoints, so I think it should be fine.

---

## Supported API Paths
1. [**GET /_api/schema**](#get-apischema) – Returns all graphs, views, and collections in the database.
2. [**GET /_api/schema/graph/&lt;graph-name&gt;**](#get-apischemagraphgraph-name) – Returns the specified graph and its connected collections.
3. [**GET /_api/schema/view/&lt;view-name&gt;**](#get-apischemaviewview-name) – Returns the specified view and its linked collections.
4. [**GET /_api/schema/collection/&lt;collection-name&gt;**](#get-apischemacollectioncollection-name) – Returns the specified collection and its schema.

> ---

Reviewer: This seems unnecessary and we don't provide such overviews anywhere in the existing content. Descriptions of endpoints should go into the OpenAPI code block.

Author: I transferred these explanations to each endpoint.

> <h2 id="get-apischema">GET /_api/schema</h2>

Reviewer: Please take a look at other existing pages. We use descriptive headlines instead of the endpoint, and raw HTML should be avoided.

         parameters:
            - name: database-name
              in: path
              required: true
              example: _system
              description: |
                The name of a database that you want to examine its structures and schemas.
              schema:
                type: string
                
            - name: sampleNum
              in: query
              required: false
              description: |
                The number of documents/edges to examine per collection when determining
                attribute types and whether an attribute is optional.
                Must be a positive integer. Defaults to `100`.
                If larger than the number of documents, it will only use the available ones.
              schema:
                type: integer
                minimum: 1
                default: 100
            
            - name: exampleNum
              in: query
              required: false
              description: |
                The number of example documents/edges to return per collection.
                Must be a non-negative integer and not be larger than `sampleNum`.

>                 If `0`, no examples will be returned. Defaults to `1`.

Reviewer: Present tense. Using `default: 1` is sufficient
```suggestion
                If set to `0`, no examples are returned.
```

              schema:
                type: integer
                minimum: 0
                default: 1
         
         responses:
            '200 OK':
               description: |
                  The schema overview was successfully returned.
            '400 Bad Request':
               description: |
                  Invalid query parameters (e.g., negative or non-numeric values).
            '404 Not Found':
               description: |
                  Collection, view or graph not found on the database.
            '405 Method Not Allowed':
               description: |
                  When request method is not GET. This endpoint only supports GET.
            '500 Internal Server Error':
               description: |

>                   Internal server error.

Reviewer: It must only be the status code, e.g. `'200':`. We don't tend to describe all of the possible status codes, especially because there can be many different ones if it's not handled by the specific rest handler but somewhere else (e.g. there is always a possibility that a request returns 503 during startup, but describing this for each endpoint would be extremely cumbersome). Which of the listed ones are specific to the schema feature?
